### ISSUE TITLE
What exactly does the LinearTokenSelector model output?

### ISSUE THREAD

**Issue by AAbathur** [complementizer/rl-sentence-compression#2]
In [scrl/model.py](https://github.com/complementizer/rl-sentence-compression/blob/main/scrl/model.py), LinearTokenSelector return F.log_softmax(x, dim=2), that means the model output log P(Y|X), is it right?
```
  class LinearTokenSelector(nn.Module):
      def __init__(self, encoder, embedding_size=768):
          super(LinearTokenSelector, self).__init__()
          self.encoder = encoder
          self.classifier = nn.Linear(embedding_size, 2, bias=False)
      def forward(self, x):
          output = self.encoder(x, output_hidden_states=True)
          x = output["hidden_states"][-1]
          x = self.classifier(x)
          x = F.log_softmax(x, dim=2)
          return x
```
While the code in line 256-257 of [scrl/training.py](https://github.com/complementizer/rl-sentence-compression/blob/main/scrl/training.py), a softmax operation is added on the model output to get P(Y|X)
```
  logits = model(input_ids)
  probs = torch.softmax(logits, dim=2) 
```

Is this a mistake or is there something I misunderstood ? Hoping for your response, thanks.

**Comment by complementizer** [comment]
Thanks for pointing out @AAbathur. At some stage I wanted to return log probs in the `forward` function. In the current state of the code this isn't necessary anymore, i.e. `LinearTokenSelector` should rather just output "raw" scores before we convert them into probabilities. Either way, this shouldn't make a difference in theory because softmax(x) is equivalent to softmax(log-softmax(x)) afaik. Just checked running the training a bit and both variants produce the same probabilities based on torch.allclose() but not torch.equal() due to tiny numerical precision differences...

I'll sanity-check this and probably remove the log-prob to avoid confusion.

**Comment by AAbathur** [comment]
OK, thanks a lot.
